### PR TITLE
[OSDOCS-12035]: Adding tenancy docs to HCP on AWS

### PR DESCRIPTION
**Pull request by lahinson**
Version(s): 4.17+

Issue: https://issues.redhat.com/browse/OSDOCS-12035

Link to docs preview: https://83719--ocpdocs-pr.netlify.app/openshift-enterprise/latest/hosted_control_planes/hcp-deploy/hcp-deploy-aws.html#hcp-aws-deploy-hc_hcp-deploy-aws

QE review:
- [x] QE has approved this change.

Additional information: Do not merge until the MCE 2.7 release on November 5

### CODE REVIEW

File: modules/hcp-aws-deploy-hc.adoc [hunk 1]
```diff
@@ -18,21 +18,22 @@ You can deploy a hosted cluster on {aws-first} by using the `hcp` command-line i
...
+    --pull-secret <path_to_pull_secret> \// <5>
+    --region <region> \// <6>
     --generate-ssh \
-    --node-pool-replicas <node_pool_replica_count> \ <7>
-    --namespace <hosted_cluster_namespace> \ <8>
-    --role-arn <role_name> <9>
+    --node-pool-replicas <node_pool_replica_count> \// <7>
+    --namespace <hosted_cluster_namespace> \// <8>
+    --role-arn <role_name> \// <9>
+    --render <10>
```
Comment (muraee) [inline modules/hcp-aws-deploy-hc.adoc:36]:
```suggestion
    --render <10> > somefile.yaml
```
you would need to redirect the output to a file when using the `render` flag, or use `--render-into somefile.yaml`

File: modules/hcp-aws-deploy-hc.adoc [hunk 2]
```diff
@@ -43,6 +44,7 @@ $ hcp create cluster aws \
 <7> Specify the node pool replica count, for example, `3`.
 <8> By default, all `HostedCluster` and `NodePool` custom resources are created in the `clusters` namespace. You can use the `--namespace <namespace>` parameter, to create the `HostedCluster` and `NodePool` custom resources in a specific namespace.
 <9> Specify the Amazon Resource Name (ARN), for example, `arn:aws:iam::820196288204:role/myrole`.
+<10> To specify whether the EC2 instance should run on shared or single-tenant hardware, include this flag.
```
Comment (muraee) [inline modules/hcp-aws-deploy-hc.adoc:47]: maybe explain what the `render` flag does first, e.g. it won't apply the k8s resources, but render them into a file instead. Then you can edit the NodePool spec in that yaml 'To specify whether the EC2 instance should run on shared or single-tenant hardware'

File: modules/hcp-aws-deploy-hc.adoc [hunk 3]
```diff
@@ -43,6 +44,9 @@ $ hcp create cluster aws \
 <7> Specify the node pool replica count, for example, `3`.
 <8> By default, all `HostedCluster` and `NodePool` custom resources are created in the `clusters` namespace. You can use the `--namespace <namespace>` parameter, to create the `HostedCluster` and `NodePool` custom resources in a specific namespace.
 <9> Specify the Amazon Resource Name (ARN), for example, `arn:aws:iam::820196288204:role/myrole`.
+<10> If you want to indicate whether the EC2 instance runs on shared or single tenant hardware, include this field. The `--render-into` flag renders Kubernetes resources into the YAML file that you specify in this field. Then, continue to the next step to edit the YAML file.
+
+. In the command to create the hosted cluster, if you included the `--render-into` flag and specified a YAML file, edit the `NodePool` specification in the YAML file to indicate whether the EC2 instance should run on shared or single-tenant hardware.
```
Comment (muraee) [inline modules/hcp-aws-deploy-hc.adoc:49]: maybe include a snippet of the yaml where the user should add the new field
```yaml
apiVersion: hypershift.openshift.io/v1beta1
kind: NodePool
metadata:
  name: <nodepool-name>
spec:
  platform:
    aws:
      placement:
        tenancy: one of "default", "dedicated" or "host"
```

Comment (lahinson) [reply modules/hcp-aws-deploy-hc.adoc:49]: Awesome. I was hoping you would provide an example like that 😄

File: modules/hcp-aws-deploy-hc.adoc [hunk 4]
```diff
@@ -43,6 +44,26 @@ $ hcp create cluster aws \
...
+  name: <nodepool_name> <1>
+spec:
+  platform:
+    aws:
+      placement:
+        tenancy: "default" <2>
+----
++
+<1> Specify the name of the `NodePool` resource.
+<2> Specify a valid value for tenancy: `"default"`, `"dedicated"`, or `"host"`.
```
Comment (heliubj18) [inline modules/hcp-aws-deploy-hc.adoc:66]: Do we need to add some explanations for those values?  e.g.
`// default: NodePool instances run on shared hardware.  `
` // dedicated: Each NodePool instance runs on single-tenant hardware.  `
` // host: NodePool instances run on user's pre-allocated dedicated hosts.`

Comment (lahinson) [reply modules/hcp-aws-deploy-hc.adoc:66]: That's helpful -- I'll add it. Thank you, @heliubj18!

File: modules/hcp-aws-deploy-hc.adoc [hunk 5]
```diff
@@ -43,6 +44,26 @@ $ hcp create cluster aws \
 <7> Specify the node pool replica count, for example, `3`.
 <8> By default, all `HostedCluster` and `NodePool` custom resources are created in the `clusters` namespace. You can use the `--namespace <namespace>` parameter, to create the `HostedCluster` and `NodePool` custom resources in a specific namespace.
 <9> Specify the Amazon Resource Name (ARN), for example, `arn:aws:iam::820196288204:role/myrole`.
+<10> If you want to indicate whether the EC2 instance runs on shared or single tenant hardware, include this field. The `--render-into` flag renders Kubernetes resources into the YAML file that you specify in this field. Then, continue to the next step to edit the YAML file.
+
+. In the command to create the hosted cluster, if you included the `--render-into` flag and specified a YAML file, edit the `NodePool` specification in the YAML file to indicate whether the EC2 instance should run on shared or single-tenant hardware.
```
Comment (michaelryanpeter) [inline modules/hcp-aws-deploy-hc.adoc:49]: I find the following easier to understand:
```suggestion
.  If you included the `--render-into` flag in the `<previous_or_preceding>` command and specified a YAML file, edit the `NodePool` specification in the YAML file. You must indicate whether the EC2 instance should run on shared or single-tenant hardware, similar to the following example:
```
As written, I read the directive as "In the command, edit the spec in the resulting YAML file, if you used the flag". Which isn't correct, right? I think you mean to say, "If you used the flag in the command, edit the spec in the resulting YAML file."

Does that make sense? Feel free to ping me if I need to clarify.

**Review by michaelryanpeter — COMMENTED** [review]
Just one comment on phrasing. Otherwise, LGTM.

/remove-label peer-review-need
/remove-label peer-review-in-progress
/label peer-review-done

File: modules/hcp-aws-deploy-hc.adoc (hunk 5 above)
Comment (lahinson) [reply modules/hcp-aws-deploy-hc.adoc:49]: That makes sense. I like your suggested revision. Thanks!

**Pull request merged**